docker/lib/dependabot/docker/file_updater.rb

```diff
@@ -109,46 +121,55 @@ def update_digest_and_tag(previous_content, old_source, new_source)
         end
       end
 
+      sig { params(source: T::Hash[Symbol, T.nilable(String)]).returns(T::Boolean) }
       def specified_with_tag?(source)
-        source[:tag]
+        !!source[:tag]
```

Comment:
Not really, it's not unheard of, but in typical Ruby code you don't care if something is explicitly true, rather that it's "truthy". In this case for sorbet it's probably the best option if we need this to return a bool

Reply:
I'd say it's common, but [not agreed upon](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/DoubleNegation).

My personal preference is to ensure that a method ending in `?` always returns `true` or `false`. To Jurre's point, Ruby itself only cares whether the value is "truthy" (i.e. not `false` or `nil`).

> Is `.nil?` not a more common approach?

`!source[:tag].nil?` might be more common. My sample size is too small to say 🙈 The caveat is that if `source[:tag] == false`, the `!source[:tag].nil?` would return `true`.

Reply:
yeah, I found that one as well, but was discouraged from using it because in the scenario that I was using it, a nil also evaluated to 'not true' so it worked just as well.
If you want to deal witht eh nil for source, though, I would use something like 
**if source[:tag].nil? raise [some exception] 
rescue [some exception]
/ handle it**